src/main/java/emissary/core/IBaseDataObject.java

```diff
@@ -350,17 +354,81 @@ enum MergePolicy {
      * 
      * @param key name of the metadata element
      * @return the string value or null if no such element
+     * @deprecated use {@link #getParameterAsConcatString(String)}
      */
-    String getStringParameter(String key);
+    @Deprecated
+    default String getStringParameter(final String key) {
+        return getParameterAsConcatString(key);
+    }
 
     /**
      * Retrieve a specified metadata element as a string value
      * 
      * @param key name of the metadata element
      * @param sep the separator for multivalued fields
      * @return the string value or null if no such element
+     * @deprecated use {@link #getParameterAsConcatString(String, String)}
+     */
+    @Deprecated
+    default String getStringParameter(final String key, final String sep) {
+        return getParameterAsConcatString(key, sep);
+    }
+
+    /**
+     * Retrieve the Collection of metadata values identified by key where each element is converted to a string
+     *
+     * @param key name of the metadata element collection
+     * @return Collection of elements converted to strings
+     */
+    default Collection<String> getParameterAsStrings(final String key) {
+        final var obj = getParameter(key);
+        if (obj == null || obj.isEmpty() || ((obj.size() == 1) && (obj.get(0) == null))) {
+            return Collections.emptyList();
+        } else if ((obj.size() == 1) && (obj.get(0) instanceof String)) {
+            return Collections.singletonList((String) obj.get(0));
+        } else {
+            return obj.stream().map(String::valueOf).collect(Collectors.toList());
+        }
+    }
+
+    /**
+     * Retrieve the metadata value identified by key where the element is converted to a string
+     *
+     * @param key name of the metadata element
+     * @return parameter converted to strings
+     * @throws IllegalStateException if the parameter has more than one value
+     */
+    default String getParameterAsString(final String key) {
+        final var obj = getParameterAsStrings(key);
+        if (obj.size() > 1) {
+            throw new IllegalStateException("There is more than one value for parameter[" + key + " ]!!");
```

Comment:
An `Exception` is probably too heavy-handed for an initial implementation, because it could cause failures that escape detection until they've been deployed.   For now, we should probably just log an error or warning so we can catalog "offending" fields.  It'll be burdensome downstream to wrap every `getParameterAsString` call with a `try...catch {log..}` block